client/state/notices/middleware.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { truncate, forEach } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { errorNotice } from 'state/notices/actions';
+import { POST_DELETE_FAILURE } from 'state/action-types';
+import { getSitePost } from 'state/posts/selectors';
+
+const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: [
```

Comment:
This is the question I raised in #6262 - it seems like the burden of proof should like in asking, "when would we have a legitimate reason to have different notices for the same type of event."